doc/user/tutorials/making_your_first_app.rst

```diff
@@ -0,0 +1,510 @@
+            setFakePower(self.r.core)
+
+
+    def setFakePower(core):
+        midplane = core[0].getHeight()/2.0
+        center = np.array([0,0,midplane])
+        peakPower = 1e6
+        mgFluxBase = np.arange(5)
+        for a in core:
+            for b in a:
+                vol = b.getVolume()
+                coords = b.spatialLocator.getGlobalCoordinates()
+                r = np.linalg.norm(abs(coords-center))
+                fuelFlag = 10 if b.isFuel() else 1.0
+                b.p.power = peakPower / r**2 * fuelFlag
+                b.p.pdens = b.p.power/vol
+                b.p.mgFlux = mgFluxBase*b.p.pdens
+
+
+Making the thermal/hydraulics kernel
+------------------------------------------
+Since we told the ARMI plugin to schedule the flux solver before thermal/hydraulics solver
+via the ``ORDER`` attribute, we can depend on there being up-to-date block-level ``power``
+state data loaded onto the ARMI reactor by the time this thermal/hydraulics solver gets
+called by the ARMI main loop.
+
+We'll make a somewhat meaningful (but still totally academic) flow solver here that uses
+energy conservation to determine an idealized coolant flow rate. To do this it will
+compute the total power produced by each assembly.
+
+.. math::
+
+    q''' = \dot{m} C_p \Delta T
+
+
+.. code-block:: python
+    :caption: ``myapp/thermalSolver.py``
+
+    from armi import interfaces
+    from armi.reactor.flags import Flags
+    from armi import runLog
+
+    # hard coded inlet/outlet temperatures
+    inletInC = 360.0
+    outletInC = 520.0
+
+    class ThermalInterface(interfaces.Interface):
+        name = "dummyTH"
+
+        def interactEveryNode(self, cycle=None, timeNode=None):
+            runLog.info("Computing idealized flow rate")
+            for assembly in self.r.core:
+                runThermalHydraulics(assembly)
+
+    def runThermalHydraulics(assembly):
+        massFlow = computeIdealizedFlow(assembly)
+        computeAxialCoolantTemperature(assembly, massFlow)
+
+    def computeIdealizedFlow(a):
+
+        # compute required mass flow rate in assembly to reach target outlet temperature
+        # mass flow rate will be constant in each axial region, regardless of coolant
+        # area (velocity may change)
+        coolants = a.getComponents(Flags.COOLANT)
+        coolantMass = sum([c.getMass() for c in coolants])
+
+        # use ARMI material library to get heat capacity for whatever the user has
+        # defined the coolant as
+        tempAvg = (outletInC + inletInC)/2.0
+        coolantProps = coolants[0].getProperties()
+        heatCapacity = coolantProps.heatCapacity(Tc=tempAvg)
+
+        deltaT = outletInC - inletInC
+        massFlowRate = a.calcTotalParam('power')/(deltaT * heatCapacity)
+        return massFlowRate
+
+    def computeAxialCoolantTemperature(a, massFlow):
+        """Compute block-level coolant inlet/outlet/avg temp and velocity."""
+        # solve q''' = mdot * Cp * dT for dT this time
```

Comment:
Include this in the docstring!

Reply:
it's in the other docstring and renders as latex math